OpenOversight/app/main/forms.py

```diff
@@ -372,4 +378,6 @@ class BrowseForm(Form):
                           validators=[AnyOf(allowed_values(AGE_CHOICES))])
     max_age = SelectField('maximum age', default=100, choices=AGE_CHOICES,
                           validators=[AnyOf(allowed_values(AGE_CHOICES))])
+    year = SelectField('year', default='', choices=YEAR_CHOICES,
+                       validators=[AnyOf(allowed_values(YEAR_CHOICES))])
```

Comment:
I think it might be better to do a `QuerySelectField` that will get the full range of years from officer employment dates, rather than using a fixed list like `YEAR_CHOICES`. For one, departments could have employees with last employment dates before 2016. Plus, we'd have to update `YEAR_CHOICES` once every year going forward. A `QuerySelectField` is nice because it will dynamically fetch the possible choices from the database. I used one in PR #619, and set the query for the field in the corresponding view function, in case that is helpful at all.

Reply:
Good point, having to each year update the hardcoded list of `YEAR_CHOICES` is a downside.  I agree about using a `QuerySelectField`, but I'm not sure the best way to represent in the database the years for which we have available data.  Here are two approaches I thought of:
1) Create a `QuerySelectField` for each department that includes every year represented in every department officer's `employment_date` or `last_employment_date`.  (This is the approach I am leaning toward.)  Don't change the officer or department model at all.
2) Add a `Years_Of_Available_Data` field to each department and list in that field the years for which we have roster data....... Then, create the `QuerySelectField` based on this new field.  I  think this approach would be overengineering.

I'm also interested in what @camfassett thinks, as she opened the original issue.

Reply:
Hey, I'm reviewing PRs that I still have pending, and I wanted to give a heads-up that I will be taking the first approach.